assets/js/app.js

```diff
@@ -0,0 +1,443 @@
+
+            // Handle modal close events
+            const modalElement = document.getElementById('replaceCoinModal');
+            modalElement.addEventListener('hidden.bs.modal', handleModalClose);
+
+            // Handle cancel button
+            document.getElementById('cancelReplace').addEventListener('click', handleModalClose);
+
+        } catch (error) {
+            console.error('Error in moreThanFiveCoins:', error);
+            pendingCoinSwitch.checked = false;
+        }
+    };
+
+
+    const handleModalClose = () => {
+        if (pendingCoinSwitch) {
+            pendingCoinSwitch.checked = false;
+        }
+        pendingCoinId = null;
+        pendingCoinSwitch = null;
+    };
+
+    const renderCoins = coinsHTML => {
+        document.getElementById("coins-container").innerHTML = coinsHTML;
+        // Initialize popovers for the buttons
+        initializePopovers();
+        // Initialize checkbox for the coins
+        initializeCheckbox();
+        // load the already selected buttons
+        loadSelectedButtons()
+    };
+
+    const generateMoreInfo = singleCoinData => {
+        const coinPriceToUSD = singleCoinData.market_data.current_price.usd;
+        const coinPriceToEUR = singleCoinData.market_data.current_price.eur;
+        const coinPriceToILS = singleCoinData.market_data.current_price.ils;
+        const coinImage = singleCoinData.image.thumb;
+
+        return `
+            <div>
+                <img src="${coinImage}" alt="${singleCoinData.name}" style="width: 50px; height: 50px;">
+                <p>Price in USD: $${coinPriceToUSD}</p>
+                <p>Price in EUR: €${coinPriceToEUR}</p>
+                <p>Price in ILS: ₪${coinPriceToILS}</p>
+            </div>
+        `;
+    };
+
+    const generateCoins = coins => {
+        return coins
+            .map(
+                coin => `
+                    <div class="card">
+                        <div class="card-body card-flex">
+                            <div class="card-title-container">
+                                <h4 class="card-title">${coin.symbol}</h4>
+                                <div class="form-check form-switch">
+                                    <input class="form-check-input" type="checkbox" id="${coin.id}Switch">
+                                    <label class="form-check-label" for="flexSwitchCheckDefault"></label>
+                                </div>
+                            </div>
+                            <p class="card-text">${coin.name}</p>
+                            <button id="${coin.id}" type="button" class="btn btn-primary btn-popover">
+                                More Info
+                            </button>
+                        </div>
+                    </div>
+                `
+            )
+            .join("");
+    };
+
+    // Update the search form event listener
+    document.getElementById('searchForm').addEventListener('submit', (event) => {
```

Comment:
once I searched for something and wants to go back to see the unfiltered list, what are my options *other than hitting search on an empty search box*?